consensus/hotstuff/signature/block_signer_decoder.go

```diff
@@ -24,8 +23,11 @@ func NewBlockSignerDecoder(committee hotstuff.DynamicCommittee) *BlockSignerDeco
 var _ hotstuff.BlockSignerDecoder = (*BlockSignerDecoder)(nil)
 
 // DecodeSignerIDs decodes the signer indices from the given block header into full node IDs.
+// Note: A block header contains a quorum certificate for its parent, which proves that
+// the block extends a valid fork. Consequently, the returned IdentifierList contains the
```

Comment:
I am not sure about the usage of "valid fork" it seems a bit misleading since this interface doesn't deal with forks in any way. Maybe we should rephrase it in a way that we discuss validity of parent? 

Something like: 
> A block header contains a quorum certificate for its parent, which proves that consensus committee has reached agreement on validity of parent block.